### ISSUE TITLE
How to convert .bin lidar files to .png

### ISSUE THREAD

**Issue by ar99-dev** [JUGGHM/PENet_ICRA2021#17]
I want to infer using your model on kitti .bin files but your model take .png files. So if it's possible to provide the code for this conversion.
Thanks in advance.

**Comment by JUGGHM** [comment]
Thanks for your interest! In this project we don't directly take .bin files as inputs. However, I have a **naive** and **incomplete** **_python_** script for converting the .bin files to .png via direct projection. Note that the projected depth maps from LiDaR pointclouds are not exactly the same as those in KITTI depth completion dataset as Uhrig et al. [3DV2017] composes ego motion compensation. I hope this script may help.
[bin2png.txt](https://github.com/JUGGHM/PENet_ICRA2021/files/6659830/bin2png.txt)

**Comment by Laihu08** [comment]
hi, @JUGGHM I tried the code what you gave for converting .bin to .png format, but I have some doubt like, why still it need to read depth from .png because this is the code to read depth .bin file and projecting in-camera coordinates and save that projection in .png file right? I would like to convert .bin to .png for depth completion in the custom dataset. Thank you very much

**Comment by JUGGHM** [comment]
> hi, @JUGGHM I tried the code what you gave for converting .bin to .png format, but I have some doubt like, why still it need to read depth from .png because this is the code to read depth .bin file and projecting in-camera coordinates and save that projection in .png file right? I would like to convert .bin to .png for depth completion in the custom dataset. Thank you very much

I just used the depth file to ensure whether the projected result is right, and you could read no depth maps actually. I didn't involve it in PENet but I think you could modify this incomplete script for your own use. The `pro_depth ` variable is the projected sparse depth in numpy format and I think it is what you want. And you might initialize it manually but not from ` #pro_depth = np.zeros_like(dc_depth)`.

**Comment by Laihu08** [comment]
Yes Exactly, because currently I am having .npy format of depth map for 3D object detection. Now can i use this .npy file in your code to estimate dense depth map and i have respective Images for those depth also but i am little confusion about calibration file because the calibration file of depth completion datasets is different from 3D object detection ! can you help me to solve this problem ?

**Comment by JUGGHM** [comment]
> Yes Exactly, because currently I am having .npy format of depth map for 3D object detection. Now can i use this .npy file in your code to estimate dense depth map and i have respective Images for those depth also but i am little confusion about calibration file because the calibration file of depth completion datasets is different from 3D object detection ! can you help me to solve this problem ?

In my script I used the calibration matrix of KITTI raw dataset, and I am not quite familiar with 3D object detection task.

**Comment by Laihu08** [comment]
oh okay thank you very much for your quick replies. can we change hyper parameter values in Loss functions while training ? because it mentioned (0,0,0) I am just curious will it work if i change it ?

**Comment by JUGGHM** [comment]
> oh okay thank you very much for your quick replies. can we change hyper parameter values in Loss functions while training ? because it mentioned (0,0,0) I am just curious will it work if i change it ?

It will work but the performance will change I think. It corresponds to the description of intermediate supervision in the implementation details in the paper.

**Comment by JUGGHM** [comment]
> oh okay thank you very much for your quick replies. can we change hyper parameter values in Loss functions while training ? because it mentioned (0,0,0) I am just curious will it work if i change it ?

I reviewed the code but found only loss items with 0, 0, 0 initialization. It is just for creating the variables.

**Comment by Laihu08** [comment]
oh maybe i maybe misunderstood the logic over there, then what value you assigned as hyper parameter for your loss functions ? Thank you!

**Comment by JUGGHM** [comment]
> oh maybe i maybe misunderstood the logic over there, then what value you assigned as hyper parameter for your loss functions ? Thank you!

`w_st1` and  `w_st2`

**Comment by Laihu08** [comment]
Can you explain more about this hyper parameter ? Thank you

**Comment by JUGGHM** [comment]
> > oh okay thank you very much for your quick replies. can we change hyper parameter values in Loss functions while training ? because it mentioned (0,0,0) I am just curious will it work if i change it ?
> 
> It will work but the performance will change I think. **It corresponds to the description of intermediate supervision in the implementation details in the paper.**

**Comment by Senwang98** [comment]
@Laihu08 
Have you solved this problem using the author's script?
I want to convert KITTI object dataset into depth too.
